File: app/src/main/java/com/mdgd/pokemon/ui/pokemon/items/PokemonProperty.java
package com.mdgd.pokemon.ui.pokemon.items;

import com.mdgd.pokemon.adapter.ViewHolderDataItem;

public interface PokemonProperty extends ViewHolderDataItem {
    int PROPERTY_IMAGE = 1;
    int PROPERTY_LABEL = 2;
    int PROPERTY_TITLE = 3;
    int PROPERTY_TEXT = 4;

    default int getNestingLevel() {
        return 0;
    }
}

File: app/src/main/java/com/mdgd/pokemon/ui/pokemon/dto/PokemonProperty.kt
package com.mdgd.pokemon.ui.pokemon.dto

import com.mdgd.pokemon.adapter.ViewHolderDataItem

interface PokemonProperty : ViewHolderDataItem {
    val nestingLevel: Int
        get() = 0

    companion object {
        const val EMPTY_VIEW = 1
        const val PROPERTY_IMAGE = 2
        const val PROPERTY_LABEL = 3
        const val PROPERTY_TITLE = 4
        const val PROPERTY_TEXT = 5
    }
}
